### PR TITLE
Python3 support for printing ConfigurationConflictError

### PR DESCRIPTION
**Pull request by janjaapdriessen**

### CODE REVIEW

**Comment by mgedmin** [comment]
LGTM

**Comment by jamadden** [comment]
I'm surprised that made it this long. That suggests it's not covered by a unit test. Would this be a good time to add one?

**Comment by janjaapdriessen** [comment]
Added a test in test_configuration_conflict_error_has_readable_exception.
Apologies about removing the whitespace cluttering the diff, my editor eats them.

**Comment by janjaapdriessen** [comment]
@jamadden @mgedmin is this PR to your satisfaction?

**Comment by jamadden** [comment]
LGTM, thanks for the test! (No worries about the whitespace, my editor does the same thing. It's easy to ignore by tacking`?w=0` onto the end of the URL.)

File: src/zope/configuration/tests/test_config.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def test_resolve_bad_dotted_import(self):
         # Import error caused by a totally wrong dotted name.
         from zope.configuration.exceptions import ConfigurationError
         c = self._makeOne()
-        exc = self.assertRaises(ConfigurationError, 
+        exc = self.assertRaises(ConfigurationError,
                           c.resolve, 'zope.configuration.nosuch.noreally')
```
Comment (tseaver) [inline src/zope/configuration/tests/test_config.py]: I find the context manager spelling of `assertRaises` much clearer, which works now that we have dropped Python 2.6 support:

```python
with self.assertRaises(ConfigurationError):
    c.resolve('zopeconfiguration.nosuch.noreally')
```

File: src/zope/configuration/tests/test_config.py [hunk 2]
```diff
@@ -1841,6 +1841,24 @@ def _b():
...
+        def _a():
+            pass
+        def _b():
+            pass
+        actions = [
+            expand_action(('a', 1), _a, includepath=('a',), info='conflict!'),
+            expand_action(('a', 1), _b, includepath=('a',), info='conflict2!'),
+        ]
+        exc = self.assertRaises(ConfigurationConflictError,
+                                self._callFUT, actions)
```
Comment (tseaver) [inline src/zope/configuration/tests/test_config.py]: To make assertions about the error message:

```python
with self.assertRaises(ConfigurationConflictError) as exc:
    self._callFUT(actions)

**Pull request merged**